pkg/controller/changetierrequest/changetierrequest_controller.go

```diff
@@ -74,5 +85,132 @@ func (r *ReconcileChangeTierRequest) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, nil
+	// if is complete, then check when status was changed delete it if the requested duration has passed
+	completeCond, found := condition.FindConditionByType(changeTierRequest.Status.Conditions, toolchainv1alpha1.ChangeTierRequestComplete)
+	if found && completeCond.Status == corev1.ConditionTrue {
+		deleted, requeueAfter, err := r.checkTransitionTimeAndDelete(reqLogger, changeTierRequest, completeCond)
+		if deleted || err != nil {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{
```

Comment:
so I assume this will work even if the operator is updated/redeployed in the interval? At startup, all `ChangeTierRequest` objects will be reconciled and a request will be requeued for the remaining time (or the time left), so that 24h after it was complete, it will be deleted. Correct?

Reply:
This is how it should work, yes.

Reply:
ok, thanks for confirming ;)

Reply:
yes exactly :-)